src/PKSim.Core/Model/RandomPopulationFactory.cs

```diff
@@ -243,6 +269,15 @@ private void perturbate(Individual currentIndividual, RandomPopulationSettings p
                originData.Height = new OriginDataParameter(createRandomValueFor(originData, populationSettings, CoreConstants.Parameters.MEAN_HEIGHT, randomGenerator, out success));
                currentIndividual.Organism.Parameter(CoreConstants.Parameters.HEIGHT).Value = originData.Height.Value;
             }
+
+            //now assign any random parameters due to disease state
+            foreach (var diseaseStateParameter in originData.DiseaseStateParameters)
+            {
+               diseaseStateParameter.Value = createDiseaseStateRandomParameterValueFor(originData, populationSettings, diseaseStateParameter, randomGenerator, out success);
+            }
+
+            if (!success) continue;
+            (success, _) = diseaseStateImplementation.IsValid(originData);
```

Comment:
Here I am also adding validation on specific case. for Instance age >=18 for CKD